pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -1599,15 +1599,22 @@ public ResultSet getSchemas() throws SQLException {
   public ResultSet getSchemas(@Nullable String catalog, @Nullable String schemaPattern)
       throws SQLException {
     String sql;
-    sql = "SELECT nspname AS \"TABLE_SCHEM\", NULL AS \"TABLE_CATALOG\" FROM pg_catalog.pg_namespace "
-          + " WHERE nspname <> 'pg_toast' AND (nspname !~ '^pg_temp_' "
-          + " OR nspname = (pg_catalog.current_schemas(true))[1]) AND (nspname !~ '^pg_toast_temp_' "
-          + " OR nspname = replace((pg_catalog.current_schemas(true))[1], 'pg_temp_', 'pg_toast_temp_')) ";
+    sql = "SELECT schema_name AS \"TABLE_SCHEM\", catalog_name AS \"TABLE_CATALOG\" FROM information_schema.schemata "
```

Comment:
as appealing as it is to use the `information schema`, this is actually a view and adds another select into this query. Is there any reason to do this ?

Reply:
Fixed